### PR TITLE
Italian translation: Chapter 3

### PR DESCRIPTION
**Pull request by sharkovsky**
Italian translation of chapter 3, as per #45

### CODE REVIEW

**Comment by sharkovsky** [comment]
@lewtun I am getting some issues with building the docs that I can't figure out. Can you help me out with this "Unexpected token" error? 
Thank you!

File: chapters/it/chapter3/4.mdx [hunk 1]
```diff
@@ -0,0 +1,359 @@
...
+
+```python out
+{'accuracy': 0.8431372549019608, 'f1': 0.8907849829351535}
+```
+
+Ancora una volta, i vostri risultati potrebbero essere leggermente diversi a causa della casualità nell'inizializzazione della testa del modello e del ricombinamento dei dati, ma dovrebbero essere nello stesso ordine di grandezza.
+
+<Tip>
+
+✏️ **Prova tu!** Modifica il ciclo di addestramento precedente per affinare il modello sul dataset SST-2.
```
Comment (EdAbati) [inline chapters/it/chapter3/4.mdx:202]: I like 'Prova tu!' more than 'Da provare!' (as in the previous pages). Nevertheless, I think we should keep them consistent :)

File: chapters/it/chapter3/6.mdx [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+
+Testiamo cosa avete imparato in questo capitolo!
+
+### 1. Il dataset `emotion` contiene messaggi Twitter etichettati con emozioni. Cercalo nel [Hub](https://huggingface.co/datasets) e leggi la carta del dataset. Quale di queste non fa parte delle sue emozioni di base?
+
+<Question
+	choices={[
+		{
+			text: "Joy (Gioia)",
+			explain: "Prova ancora — questa emozioni è contenuta nel dataset!"
```
Comment (EdAbati) [inline chapters/it/chapter3/6.mdx:15]: Typo 'emozionE'

File: chapters/it/chapter3/6.mdx [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+		},
+		{
+			text: "Love (Amore)",
+			explain: "Prova ancora — questa emozioni è contenuta nel dataset!"
```
Comment (EdAbati) [inline chapters/it/chapter3/6.mdx:19]: Typo same as above 'emozionE'

**Review by lewtun — APPROVED** [review]
Thank you for this epic translation @sharkovsky and welcome to the 🤗 community! I think the error was coming from an extra line in the quiz, so I've fixed that and will merge once the CI is green - thanks again!

File: chapters/it/chapter3/6.mdx [hunk 4]
```diff
@@ -0,0 +1,296 @@
...
+			explain: "Questo è sicuramente uno dei benefici di questo metodo! Ma non è l'unico...",
+            correct: true
+		},
+	]}
+/>
+
+### 5. Qual è il significato di padding dinamico (dynamic padding)?
+
+<Question
+	choices={[
```
Comment (EdAbati) [inline chapters/it/chapter3/6.mdx:107]: I would change the 3 answers adding the verb at the beginning: 'È quando si...'

File: chapters/it/chapter3/6.mdx [hunk 5]
```diff
@@ -0,0 +1,296 @@
...
+		{
+			text: "Quando si applica in ogni batch il padding fino alla lunghezza massima dell'intero dataset",
+			explain: "Coinvolge il padding per ogni batch, ma non fino alla lunghezza massima dell'intero dataset."
```
Comment (EdAbati) [inline chapters/it/chapter3/6.mdx:110]: I would change 'coinvolge' with 'riguarda'

**Comment by sharkovsky** [comment]
Thank you @lewtun! It's been a lot of fun to translate chapter 3 :)

File: chapters/it/chapter3/6.mdx [hunk 6]
```diff
@@ -0,0 +1,296 @@
...
+			text: "Quando si applica il padding agli input in modo che ogni frase abbia lo stesso numero di token della frase precedente nel dataset.",
+			explain: "Sbagliato. In più non avrebbe senso considerare l'ordine del dataset dato che lo rimischiamo durante l'apprendimento."
+		},
+	]}
+/>
+
+### 6. Qual è lo scopo di una funzione di raccolta (collate function)?
+
+<Question
+	choices={[
```
Comment (EdAbati) [inline chapters/it/chapter3/6.mdx:127]: I would change the preposition at the beginning of all these answers: 'Di' instead of 'per'. E.g. ' Di raccogliere...' etc.

**Review by EdAbati — COMMENTED** [review]
Hey, I had a look at this. It looks good :) 
I mainly found some typos, potential small formatting issues and some of your personal comments that you left in some points.

I have also added some minor suggestions on how to change certain translation. Feel free to  skip these if you don't agree.

I am not on my laptop and I could  not check your failing test, but I could try to reproduce it next week in case it's still unsolved. :)

**Comment by sharkovsky** [comment]
Thank you for the great review @EdAbati !

@lewtun I've marked the PR as WIP in the hope that it won't be merged, please give me a couple of days to implement @EdAbati suggestions. I'll ping you when it's ready again. 

Thank you again!

**Comment by lewtun** [comment]
> @lewtun I've marked the PR as WIP in the hope that it won't be merged, please give me a couple of days to implement @EdAbati suggestions. I'll ping you when it's ready again.

Sure! Maybe ping @lvwerra when you're as I'll be on vacation for the next couple of weeks :)

**Comment by lewtun** [comment]
Hey @sharkovsky - I'm back from vacation, so happy to review this again whenever you're ready :)

**Comment by sharkovsky** [comment]
Thank you @lewtun ! Please give me a couple more weeks as I'm currently on vacation , and unfortunately I haven't had a chance to implement @EdAbati 's suggestions !

**Comment by lewtun** [comment]
> Thank you @lewtun ! Please give me a couple more weeks as I'm currently on vacation , and unfortunately I haven't had a chance to implement @EdAbati 's suggestions !

No rush - enjoy your vacation 🤗

File: chapters/it/chapter3/1.mdx [hunk 7]
```diff
@@ -0,0 +1,21 @@
+<FrameworkSwitchCourse {fw} />
+
+# Introduzione
+
+Nel [Capitolo 2](/course/chapter2) abbiamo scoperto come utilizzare i tokenizzatori e i modelli preaddestrati per effettuare delle predizioni. Ma cosa fare se si vuole affinare un modello preaddestrato col tuo insieme di dati? Lo scopriremo in questo capitolo! Imparerete:
```
[the remark this answers is not in the document]
Comment (sharkovsky) [reply chapters/it/chapter3/1.mdx:5]: I used 1PP whenever possible, and infinitive otherwise, as agreed on the forum. Thanks for spotting this!

File: chapters/it/chapter3/2.mdx [hunk 8]
```diff
@@ -0,0 +1,386 @@
...
+    test: Dataset({
+        features: ['sentence1', 'sentence2', 'label', 'idx'],
+        num_rows: 1725
+    })
+})
+```
+
+Il risultato è un oggetto di tipo `DatasetDict` che contiene il training set, il validation set, e il test set. Ciascuno di questi contiene svariate colonne, (`sentence1`, `sentence2`, `label`, e `idx`) a un numero variabile di righe, corrispondenti al numero di elementi in ogni set (quindi, vi sono 3668 coppie di frasi nel training set, 408 nel validation set, e 1725 nel test set).
+
+Questo comando scarica il dataset e lo mette in cache, in *~/.cache/huggingface/dataset* secondo l'impostazione predefinita. Nel Capitolo 2 è stato spiegato come personalizzare la cartella di cache valorizzando la variabile d'ambiente `HF_HOME`.
```
[the remark this answers is not in the document]
Comment (sharkovsky) [reply chapters/it/chapter3/2.mdx:119]: I agree. Changed to 'impostando'

File: chapters/it/chapter3/2.mdx [hunk 9]
```diff
@@ -0,0 +1,386 @@
...
+
+Ogni coppia di frasi nell'ogetto `raw_datasets` può essere ottenuto tramite indexing, come in un dizionario:
```
[the remark this answers is not in the document]
Comment (sharkovsky) [reply chapters/it/chapter3/2.mdx:121]: changed to ottenuta tramite il suo indice

File: chapters/it/chapter3/2.mdx [hunk 10]
```diff
@@ -0,0 +1,386 @@
...
+
+### Preprocessing del dataset
+
+{#if fw === 'pt'}
+<Youtube id="0u3ioSwev3s"/>
+{:else}
+<Youtube id="P-rZWqcB6CE"/>
+{/if}
+
+Per preprocessare il dataset, è necessario convertire il testo in numeri comprensibili al modello. Come dimostrato nel [capitolo precedente]/course/chapter2), ciò viene fatto con un tokenizzatore. Il tokenizer prende come input sia una frase sia una lista di frasi, quindi è possibile effettuare la tokenizzazione di tutte le prime e seconde frasi di ogni coppia in questo modo:
```
[the remark this answers is not in the document]
Comment (sharkovsky) [reply chapters/it/chapter3/2.mdx:165]: I followed your suggestion to keep `tokenizer` throughout the whole document and explain it with (tokenizzatore) the first time it appears.

**Comment by sharkovsky** [comment]
Hi @lewtun, I have implemented @EdAbati 's suggestions. (By the way, I had not realized what an epic re-reading effort you had made. Kudos for your attention to detail and thank you so much!)

On my end everything would be ready, but I see I'm failing an automatic check (not sure why). Maybe you could help me sort this out? 

Thanks!

**Comment by lewtun** [comment]
> On my end everything would be ready, but I see I'm failing an automatic check (not sure why). Maybe you could help me sort this out?

Awesome work with integrating all the suggestions @sharkovsky ! I just ran `make style` to fix some code formatting and this should be good to merge once the CI is green

Thank you again!

**Comment by sharkovsky** [comment]
@lewtun sorry to bother you, I noticed my name is missing from italian translators in the front-page README.md .... Is this intended? 
Thank you!
